state.go

```diff
@@ -215,10 +215,14 @@ func (m *Memberlist) probeNode(node *nodeState) {
 
 	// Setup an ack handler
 	ackCh := make(chan bool, m.config.IndirectChecks+1)
-	m.setAckChannel(ping.SeqNo, ackCh, m.config.ProbeInterval)
+	sent := time.Now()
+	m.setAckChannel(ping.SeqNo, ackCh, func(payload []byte) {
```

Comment:
This is a bit dirty and works around the indented use of an ack channel. Lets add a new type `ackResponse` that includes the `Payload` and `Complete` bool, and change the type of the channel to `ackResponse` instead of bool. This way we aren't overloading by injecting a second handler.